src/speculators/data_generation/preprocessing.py

```
@@ -267,17 +271,20 @@ def _preprocess_batch(
 
     results: dict[str, list] = {"input_ids": [], "loss_mask": [], "seq_len": []}
     conversations = examples.get("conversations", [])
+    metadatas= examples.get("metadata", [])
 
     if not conversations:
         log.warning(f"No conversations key found. Keys: {list(examples.keys())}")
         return results
 
-    for idx, conv in enumerate(conversations):
+    for idx, (conv,metadata) in enumerate(zip(conversations,metadatas)):
```

Ruff (0.15.13)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)